h2o-core/src/main/java/water/util/ModelPngWriter.java

```diff
@@ -0,0 +1,81 @@
+package water.util;
+
+import hex.Model;
+import water.api.StreamWriter;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class ModelPngWriter implements StreamWriter {
+
+    private final Model model;
+
+    public ModelPngWriter(Model model) {
+        this.model = model;
+    }
+
+    @Override
+    public void writeTo(OutputStream os) {
+        try {
+            Path pngtemp = Files.createTempFile("", "visualization.png");
+            model.visualize(pngtemp.toAbsolutePath().toString());
+
+            try (InputStream is = new FileInputStream(writeToZip(pngtemp).toFile())) {
+                byte[] buf = new byte[8192];
+                int c;
+                while ((c = is.read(buf, 0, buf.length)) > 0) {
+                    os.write(buf, 0, c);
+                    os.flush();
+                }
+                os.close();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Creates new ZIP archive containing file or directory represented by passed {@code path}. If {@code path}
+     * represents directory, all its files and sub-directories are recursively added to the resulting archive.
+     *
+     * @param path file or directory to be included in new ZIP archive
+     * @return {@link Path} of newly created ZIP file
+     */
+    private Path writeToZip(Path path) throws IOException {
```

Comment:
I think we have code like this around the codebase already, don't we? maybe we should reuse it?

Reply:
will look for it

Reply:
What we have is not directly usable - it is made specifically for logs archiving (LogArchiveWriter.java). Probably it would be possible to extend existing ZipUtil with archive creation functionality and try to use them in both places. At this time, there are only some methods for zip archive reading. Nevertheless, I would rather create separate task for it and do it independently of this.